openpype/hosts/maya/api/lib.py

```diff
@@ -3129,31 +3137,46 @@ def remove_render_layer_observer():
         pass
 
 
-def update_content_on_context_change():
-    """
-    This will update scene content to match new asset on context change
+def update_instances_frame_range():
+    """Update 'frameStart', 'frameEnd', 'handleStart', 'handleEnd' and 'asset'
+    attributes of sets that got one, except if instance family is 'render'
     """
-    scene_sets = cmds.listSets(allSets=True)
+    collected_instances = cmds.ls(
+        "*.id",
+        long=True,
+        type="objectSet",
+        recursive=True,
+        objectsOnly=True
+    )
     asset_doc = get_current_project_asset()
-    new_asset = asset_doc["name"]
-    new_data = asset_doc["data"]
-    for s in scene_sets:
-        try:
-            if cmds.getAttr("{}.id".format(s)) == "pyblish.avalon.instance":
-                attr = cmds.listAttr(s)
-                print(s)
-                if "asset" in attr:
-                    print("  - setting asset to: [ {} ]".format(new_asset))
-                    cmds.setAttr("{}.asset".format(s),
-                                 new_asset, type="string")
-                if "frameStart" in attr:
-                    cmds.setAttr("{}.frameStart".format(s),
-                                 new_data["frameStart"])
-                if "frameEnd" in attr:
-                    cmds.setAttr("{}.frameEnd".format(s),
-                                 new_data["frameEnd"],)
-        except ValueError:
-            pass
+
+    asset_data = asset_doc["data"]
+    frames_attributes = {
+        'frameStart': asset_data["frameStart"],
+        'frameEnd': asset_data["frameEnd"],
+        'handleStart': asset_data["handleStart"],
+        'handleEnd': asset_data["handleEnd"],
+        'asset': asset_doc['name']
```

Comment:
**Wait a second!** I had totally missed this but this to me is confusing behavior now with this function name and it triggering on `reset_frame_range`. I would not expect _that_ to force reset the asset. We should probably change this, right?

@tokejepsen @antirotor Thoughts?

Reply:
Yeah, that goes outside the scope of this method.
Also since we are looking at Maya here specifically we dont need to reset the asset cause that is validated by https://github.com/ynput/OpenPype/blob/develop/openpype/hosts/maya/plugins/publish/validate_instance_in_context.py.
An OT question is whether we want to elevate the above plugin to be more universal. Especially with the new publisher, it would be very easy to make it universal (i think).